### PR TITLE
test: increasing selected-location-factory tests

### PR DESCRIPTION
**Pull request by twitchett**
Upon select, the available options of the location drop down menu remain filtered.

Closes https://jira.ehealthafrica.org/browse/CCSL-1930

### CODE REVIEW

**Comment by danse** [comment]
Should we update the changelog here?

**Comment by danse** [comment]
I mean, it was more relevant when this library was shared between different projects, but if we don't want to keep it anymore then we should add a deprecation note or something

**Comment by danse** [comment]
Tests seem to fail here for me

**Comment by danse** [comment]
If tests are actually failing, i recommend using test driven development here. This is one of the few components where we can do that, and it pays off. There is already a pretty comprehensive unit test suite

**Comment by twitchett** [comment]
@danse looks like we finally got to the bottom of this

File: test/unit/eHealth.locations/services/selected-location-factory.js [hunk 1]
```diff
@@ -94,39 +99,63 @@ describe('Service: SelectedLocationFactory', function () {
...
       });
+      describe('when a lower level without selecting the parent levels first', function () {
+        beforeEach(function () {
+          selectFirst(THIRD)
+        });
+        it('filters the higher level', function () {
+          expect(location.levels[SECOND].items.length).toBe(8);
+        });
+        // skip test: fix not yet implemented
+        xit('filters the current level', function () {
```
Comment (danse) [inline test/unit/eHealth.locations/services/selected-location-factory.js]: how about this test? Did you forget it, or you plan to add something?

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this
 file. This file is structured according to http://keepachangelog.com/
 
 - - -
+## 4.2.1 - 2017-01-31
```
Comment (danse) [inline CHANGELOG.md]: Wait you are not changing the code here, just the tests, so nothing should change in the release notes, right?

File: test/unit/eHealth.locations/services/selected-location-factory.js (hunk 1 above)
Comment (twitchett) [inline test/unit/eHealth.locations/services/selected-location-factory.js]: Nope, I'm using your TDD approach :p 
Additional bugs originating in this repo were discovered, so I wrote tests for them. Tickets to fix them are being created in CC.

File: CHANGELOG.md (hunk 2 above)
Comment (twitchett) [inline CHANGELOG.md]: Right - I forgot to remove that

File: test/unit/eHealth.locations/services/selected-location-factory.js (hunk 1 above)
Comment (danse) [inline test/unit/eHealth.locations/services/selected-location-factory.js]: Aha ... so there was something to fix both in the call centre and in here. Alright cool

**Pull request merged**